File: monolithic/backend/app/admin/service/internal/service/position.go
package service

import (
	"context"

	"github.com/go-kratos/kratos/v2/log"
	"google.golang.org/protobuf/types/known/emptypb"

	"kratos-monolithic-demo/app/admin/service/internal/data"

	pagination "github.com/tx7do/kratos-bootstrap/gen/api/go/pagination/v1"
	userV1 "kratos-monolithic-demo/gen/api/go/user/service/v1"
)

type PositionService struct {
	userV1.UnimplementedPositionServiceServer

	log *log.Helper

	uc *data.PositionRepo
}

func NewPositionService(uc *data.PositionRepo, logger log.Logger) *PositionService {
	l := log.NewHelper(log.With(logger, "module", "position/service/admin-service"))
	return &PositionService{
		log: l,
		uc:  uc,
	}
}

func (s *PositionService) ListPosition(ctx context.Context, req *pagination.PagingRequest) (*userV1.ListPositionResponse, error) {
	return s.uc.List(ctx, req)
}

func (s *PositionService) GetPosition(ctx context.Context, req *userV1.GetPositionRequest) (*userV1.Position, error) {
	return s.uc.Get(ctx, req)
}

func (s *PositionService) CreatePosition(ctx context.Context, req *userV1.CreatePositionRequest) (*emptypb.Empty, error) {
	err := s.uc.Create(ctx, req)
	if err != nil {

		return nil, err
	}

	return &emptypb.Empty{}, nil
}

func (s *PositionService) UpdatePosition(ctx context.Context, req *userV1.UpdatePositionRequest) (*emptypb.Empty, error) {
	err := s.uc.Update(ctx, req)
	if err != nil {

		return nil, err
	}

	return &emptypb.Empty{}, nil
}

func (s *PositionService) DeletePosition(ctx context.Context, req *userV1.DeletePositionRequest) (*emptypb.Empty, error) {
	_, err := s.uc.Delete(ctx, req)
	if err != nil {
		return nil, err
	}

	return &emptypb.Empty{}, nil
}

File: monolithic/frontend/src/api/app/position.ts
import client from "/@/api/app/index";
import {components} from "/#/openapi";

// 获取列表
export const ListPosition = async (params: any) => {
    return client.GET("/admin/v1/positions",
        {
            errorMessageMode: 'none',
        },
        {params: params}
    );
};

// 获取
export const GetPosition = (params: any) => {
    return client.GET("/admin/v1/positions/{id}",
        {
            errorMessageMode: 'none',
        },
        {params: params}
    )
};

// 创建
export const CreatePosition = (params: components["schemas"]["CreatePositionRequest"]) => {
    return client.POST("/admin/v1/positions",
        {
            errorMessageMode: 'none',
        },
        {data: params}
    )
};

// 更新
export const UpdatePosition = (params: components["schemas"]["Position"]) => {
    return client.PUT("/admin/v1/positions/{id}",
        {
            errorMessageMode: 'none',
        },
        {
            params: {
                path: {
                    id: params.id!,
                }
            },
            data: params
        }
    )
};

// 删除
export const DeletePosition = (params: any) => {
    return client.DELETE("/admin/v1/positions/{id}",
        {
            errorMessageMode: 'none',
        },
        {
            params: {
                path: {
                    id: params.id,
                }
            },
        }
    )
};
